### ISSUE TITLE
Cannot run the code

### ISSUE THREAD

**Issue by qingyong-hu** [Joostytibo/EVAL-ADXL355Z#1]
Hi! I am trying to use Arduino to read data from EVAL-ADXL355Z. Thanks for your code! However, I cannot get results from the ADXL355. Specifically, the code loops in the dataReady() function. Could you tell me how you make it work?

**Comment by qingyong-hu** [comment]
From the enable() function, I print the return value of Wire.endTransmission(), the returned value is 2. I think maybe my ADXL355 did not connect to the arduino well.

**Comment by Joostytibo** [comment]
Hi, thanks for your comment.
I've struggled for quite a long time to make it work (connection etc.) to get the 'Read Data' results too.
I'm not a software engineer nor am I very experienced with coding, but the Arduino code I made works well for me.
Several other codes on Github or the Arduino site didn't work for me, so I decided to write this code.

> the returned value is 2

Maybe this has something to do with the gravition that has been set on "RANGE_2G"?


Can you give me a screenshot from the data that you received on your serial monitor perhaps?

The picture underneath shows my wiring to make the connection to work properly:
<img width="214" alt="EVAL-ADXL355Z" src="https://user-images.githubusercontent.com/74665616/116312314-928f2400-a7ac-11eb-8609-f98a8f8ffaab.PNG">

Kind regards,

**Comment by Huqy518** [comment]
Thanks so much for your kind reply!
I just put ADXL355 on the table, maybe the biggest gravition is within RANGE_2G?
The output of my serial monitor is as follows:
![image](https://user-images.githubusercontent.com/26104574/116354718-65ce2180-a82b-11eb-9018-7255b6d40316.png)
And my wiring is 
![image](https://user-images.githubusercontent.com/26104574/116355202-1a684300-a82c-11eb-9999-826b310e05a5.png)

Thanks so much for your help!

**Comment by qingyong-hu** [comment]
Thanks so much for your kind reply!
I just put ADXL355 on the table, maybe the biggest gravition is within RANGE_2G?
The output of my serial monitor is as follows:
![image](https://user-images.githubusercontent.com/26104574/116355414-66b38300-a82c-11eb-909e-b6f4f7b478cc.png)

And my wiring is
![1326929c421c8a9ccae8193732eaa2e](https://user-images.githubusercontent.com/26104574/116355462-78952600-a82c-11eb-9033-56250abd6705.jpg)

Thanks so much for your help!

**Comment by qingyong-hu** [comment]
A4 is connected to SDA (Header P2, 6)
A5 is connected to SCL  (Header P2, 2)
3.3v output of Arduino is connected to VDD (Header P1, 3)
GND of Arduino is connected to GND on the ADXL355 (Header P1, 5)

**Comment by Joostytibo** [comment]
Hi, sorry for the late reply.

I used an Arduino UNO for my ADXL355, but that shouldn't be different from an Arduino MEGA(?).
I'm going to try it out with an Arduino Mega 2560 tomorrow. I need to solder the pin headers on my new ADXL355 first, then I'll let you know if it works with this code or not.
The wiring looks fine, however I would prefer to use smaller wires for the I2C connection.
Large wires can mess with the connection, that's why I used shorter ones.

If you want to make sure that you've the right address and a good connection, you can use an 'I2C-Scanner' like this one:
[https://create.arduino.cc/projecthub/abdularbi17/how-to-scan-i2c-address-in-arduino-eaadda](url)

I know it's a struggle, I've been through it all in around 2 months, but eventually the results were there. I used some sites to understand I2C better and how to work with ADXL355. I'll also provide some links to some questions that I've asked on some websites.

[https://howtomechatronics.com/tutorials/arduino/how-i2c-communication-works-and-how-to-use-it-with-arduino/](url)
[https://www.digikey.com/en/articles/why-the-inter-integrated-circuit-bus-makes-connecting-ics-so-easy](url)

The questions:
[https://ez.analog.com/mems/f/q-a/538953/accelerometer-adxl355-moved-values](url)
[https://forum.arduino.cc/t/solved-accelerometer-adxl355-moved-values/686179](url)

I'm working from home at the moment, so all the stuff like my 'old' ADXL355 (from the picture above) is at my company.
Like I said before, I need to solder the pin headers first, then I can check if the code works properly on my Arduino MEGA.

Let me know if you need something else.

Kind regards.

**Comment by Joostytibo** [comment]
2/2

> maybe the biggest gravition is within RANGE_2G?

If I'm right, the values within RANGE_2G should be +1 and -1, when it's flat it should be 0 (like +1 + -1 = 0). But I'm not sure.

Picture underneath: The gravitation values on the left are the ones from the datasheet and the ones on the right are my values. Maybe this will help you a little.
![EVAL-ADXL355_Values](https://user-images.githubusercontent.com/74665616/116483697-e28ee900-a887-11eb-8058-0eab45a798ae.PNG)

P.S. If the links above don't work, make the line blue, right mouse click and then "go to" on that tab.

Kind regards.

**Comment by qingyong-hu** [comment]
Hi! Thanks so much for your kind reply!!!!
I have figured it out yesterday. 
There are three reasons:
1. The I2C ports in Arduino MEGA2560 are different from those in Arduino UNO. In MEGA2560, the SCL is 21 and SDA is 20.
2. Not only VDD is needed, but also VDDIO is needed to connect. Thus, I use a breadboard to make a parallel connection and produce two 3.3V outputs of Arduino board.
3. Some wires I use may not work well. I changed the wires and found that.
With these actions the output of XYZ is normal. When I put it on the table, X is 0, Y is 0, Z is 1 approximately (because the table is not so flat).
In all, thanks so much for your code and kind help! You really help me a lot!!!!
